jsx/src/components/DynamicTable/DynamicTable.jsx

```diff
@@ -0,0 +1,244 @@
+      message: "",
+      message2: "",
+      propobject: propobject,
+    });
+  }
+
+  handleValueChanged(i, event) {
+    var propvalues = this.state.propvalues;
+    var propkeys = this.state.propkeys;
+    propvalues[i] = event.target.value;
+
+    this.handleRefresh();
+    this.setPropKeys(propkeys);
+    this.setPropValues(propvalues);
+    this.setState({
+      propvalues: propvalues,
+    });
+  }
+  handleKeyChanged(i, event) {
+    var propkeys = this.state.propkeys;
+
+    if (event.target.value != "") {
+      propkeys[i] = event.target.value;
+    }
+    console.log(event.target.value);
+
+    if (event.target.value == "") {
+      this.handleItemDeleted(i);
+    }
+    this.handleRefresh(i);
+    this.setPropKeys(propkeys);
+    this.setState({
+      propkeys: propkeys,
+    });
+  }
+
+  handleItemDeleted(i) {
+    var propvalues = this.state.propvalues;
+    var propkeys = this.state.propkeys;
+
+    propvalues.splice(i, 1);
+    propkeys.splice(i, 1);
+    this.setPropKeys(propkeys);
+    this.setPropValues(propvalues);
+    this.handleRefresh(i);
+    this.setState({
+      propvalues: propvalues,
+      propkeys: propkeys,
+    });
+  }
+
+  renderKeyRows() {
+    var context = this;
+
+    return this.state.propkeys.map(function (o, i) {
+      return (
+        <tr key={"item-" + i}>
+          <td>
+            <input
+              className="form-control"
+              type="text"
+              value={o}
+              id={o + i}
+              onChange={context.handleKeyChanged.bind(context, i)}
+            />
+          </td>
+        </tr>
+      );
+    });
+  }
+  renderValueRows() {
+    var context = this;
+
+    return this.state.propvalues.map(function (o, i) {
+      return (
+        <tr key={"item-" + i}>
+          <td>
+            <input
+              className="form-control"
+              type="text"
+              value={o}
+              onChange={context.handleValueChanged.bind(context, i)}
+            />
+          </td>
+        </tr>
+      );
+    });
+  }
+  renderDelete() {
+    var context = this;
+
+    return this.state.propvalues.map(function (o, i) {
+      return (
+        <tr key={"item-" + i}>
+          <td>
+            <button
+              className="btn btn-default"
+              onClick={context.handleItemDeleted.bind(context, i)}
+            >
+              Delete
+            </button>
+          </td>
+        </tr>
+      );
+    });
+  }
+
+  render() {
+    return (
+      <div>
+        <table className="">
+          <thead>
+            <tr>
+              <th>Key</th>
+              <th>Value</th>
+            </tr>
+          </thead>
+          <tbody>
+            <tr>
+              <td>{this.renderKeyRows()}</td>
+              <td>{this.renderValueRows()}</td>
+              <td>{this.renderDelete()}</td>
+            </tr>
+          </tbody>
+        </table>
+        <form>
+          <tr>
+            <td>
+              <input
```

Comment:
Could you try to align this input/button combination? I think there's a way to "join" them together. Check out the user input.